includes/user-functions.php

```diff
@@ -639,3 +639,19 @@ function give_get_donor_name_by( $id = 0, $from = 'donation' ) {
 	return trim( $name );
 
 }
+
+/**
+ * Checks whether the given donor email exists in users as well as additional_email of donors.
+ *
+ * @since 1.8.9
+ *
+ * @param  string       $email Donor Email.
+ * @return int|false    The user's ID on success, and false on failure.
+ */
+function give_donor_email_exists( $email ) {
+	if ( Give()->donors->get_donor_by( 'email', $email ) || email_exists( $email ) ) {
```

Comment:
@mehul0810 as we discussed take out `email_exists` function from this. Basically, we can have a donor and WordPress user with same email address ( which are not connected to each other). this function only check for donor email